src/components/UsersGrid/UsersGrid.jsx

```diff
@@ -0,0 +1,82 @@
+import { Component } from 'react';
+import PropTypes from 'prop-types';
+import classes from './UsersGrid.module.css';
+import ModeButtons from '../ModeButtons/ModeButtons';
+import { UniversalButton } from '../Buttons/UniversalButton/UniversalButton';
+import GridRow from '../GridRow/GridRow';
+import Modal from '../Modal/Modal';
+import Form from '../Form/Form';
+import roles from '../../dictionary/roles';
+
+class UsersGrid extends Component {
+  constructor(props) {
+    super(props);
+    this.data = props.data;
+    this.state = {
+      userData: this.data,
+      modalIsActive: false,
+    };
+    this.openModal = this.openModal.bind(this);
+    this.closeModal = this.closeModal.bind(this);
+  }
+
+  openModal() {
+    this.setState({
+      modalIsActive: true,
+    });
+  }
+
+  closeModal() {
+    this.setState({
+      modalIsActive: false,
+    });
+  }
+
+  render() {
+    const { userData, modalIsActive } = this.state;
+    const userGridHeaderNames = [
+      { label: '#', type: 'div' },
+      { label: 'Name', type: 'div' },
+      { label: 'Age', type: 'div' },
+      { label: 'Role', type: 'div' },
+      { label: 'Action', type: 'div' },
+    ];
+    const userGridHeaderRow = (
+      <GridRow rowItemsInfo={userGridHeaderNames} rowClassName={classes.row} cellClassName={classes.cell} />
+    );
+    const usersGridRowList = userData.map(({ id, firstName, lastName, age, isAdmin }, index) => {
+      const rowItemsInfo = [
+        { label: index + 1, type: 'div' },
+        { label: `${firstName} ${lastName}`, type: 'div' },
+        { label: age, type: 'div' },
+        { label: isAdmin ? roles.ADMIN : roles.USER, type: 'div' },
+        { label: <ModeButtons {...userData[index]} />, type: 'div' },
```

Comment:
component <Label />

Reply:
I created separate function for this logics.
```suggestion
const TasksGrid = ({ data, onEditTask, onDeleteTask, onShowDetails }) => {
  const gridBodyRowsData = getGridBodyRowsData(data, onEditTask, onDeleteTask, onShowDetails);

  return <TableGrid headerRow={gridHeaderNames} bodyRows={gridBodyRowsData} />;
};
```